### ISSUE TITLE
Duplicate entries silently ignored in entites JSON deserialization

### ISSUE THREAD

**Issue by john-h-kastner-aws** [cedar-policy/cedar#357]
- [x] Duplicate attributes

```rust
    #[test]
    fn json_duplicate_attributes_error() {
        let json = r#"[
            {
                "uid": { "type": "foo", "id": "bar" },
                "attrs": {
                    "bacon": "eggs",
                    "bacon": "beans"
                },
                "parents": []
            }
        ]"#;
        assert_deserialization_err(json, "");
    }

    fn assert_deserialization_err(json: &str, err: &str) {
        let eparser: EntityJsonParser<'_> =
            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
        assert_matches!(eparser.from_json_str(json), Err(EntitiesError::Deserialization(actual_err)) => {
                assert!(actual_err.to_string().contains(err), "Expected an error containing `{}`, but saw `{}`", err, actual_err);
        });
    }
```

- [x] Duplicate UIDs (#159)

```rust
    #[test]
    fn json_duplicate_uid_error() {
        let json = r#"[
            { "uid": { "type": "foo", "id": "bar" }, "attrs": {}, "parents": [] },
            { "uid": { "type": "foo", "id": "bar" }, "attrs": {}, "parents": [] }
        ]"#;
        assert_deserialization_err(json, "");
    }
```

**Comment by hakanson** [comment]
add this test case data which uses the different options to specify the uid

```json
[
  {
    "uid": "Test::User::\"user1\"",
    "attrs": {},
    "parents": []
  },
  {
    "uid": { "type": "Test::User", "id": "user1" },
    "attrs": {},
    "parents": []
  },
  {
    "uid": {
      "__entity": { "type": "Test::User", "id": "user1" }
    },
    "attrs": {},
    "parents": []
  }
]
```

**Comment by shaobo-he-aws** [comment]
Confirmed these examples (including @hakanson's) are handled properly. Probably fixed by #365.
